### PR TITLE
Add final states for Authorization view

### PR DESCRIPTION
**Pull request by ConstantinKV**
https://github.com/saltedge/sca-authenticator-android/issues/12

### CODE REVIEW

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/details/AuthorizationDetailsFragment.kt [hunk 1]
```diff
@@ -110,17 +102,18 @@ class AuthorizationDetailsFragment : BaseFragment(),
...
-        } }
+    override fun onClick(view: View?) {
+        presenterContract.onViewClick(view?.id ?: return)
+    }
+
+    override fun getUpActionImage(): Int? = R.drawable.ic_close_white_24dp
 
-        contentView?.setTitle(presenterContract.title)
-        contentView?.setDescription(presenterContract.description)
+    override fun updateViewsContent() {
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/details/AuthorizationDetailsFragment.kt]: Remove please this method

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/details/AuthorizationDetailsFragment.kt [hunk 2]
```diff
@@ -110,17 +102,18 @@ class AuthorizationDetailsFragment : BaseFragment(),
...
-        headerView?.setLogoUrl(presenterContract.providerLogo)
-        headerView?.setTitle(presenterContract.providerName)
-        presenterContract.startTime?.let { startTime -> presenterContract.endTime?.let { endTime ->
-            headerView?.setProgressTime(startTime, endTime)
-        } }
+    override fun onClick(view: View?) {
+        presenterContract.onViewClick(view?.id ?: return)
+    }
+
+    override fun getUpActionImage(): Int? = R.drawable.ic_close_white_24dp
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/details/AuthorizationDetailsFragment.kt]: Mb better `getUpActionImageResId()`?

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/AuthorizationViewModel.kt [hunk 3]
```diff
@@ -40,45 +40,71 @@ data class AuthorizationViewModel(
...
+     * @return true if expiresAt time is before now
+     */
+    val isExpired: Boolean
+        get() = expiresAt.isEqualNow || expiresAt.isBeforeNow
+
+    /**
+     * Check what authorization is not expired
+     *
+     * @receiver authorization view model
+     * @return true if expiresAt time is after now
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/AuthorizationViewModel.kt]: @return boolean, true if expiresAt time is after now

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/AuthorizationViewModel.kt [hunk 4]
```diff
@@ -98,11 +124,19 @@ fun AuthorizationData.toAuthorizationViewModel(connection: ConnectionAbs): Autho
         expiresAt = this.expiresAt,
         createdAt = this.createdAt ?: DateTime(0L),
         authorizationID = this.id,
-        authorizationCode = this.authorizationCode ?: "",
-        isProcessing = false
+        authorizationCode = this.authorizationCode ?: ""
     )
 }
 
+fun List<AuthorizationViewModel>.joinFinalModels(listWithFinalModels: List<AuthorizationViewModel>): List<AuthorizationViewModel> {
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/AuthorizationViewModel.kt]: Can u please add annotation for this method?

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/BaseAuthorizationPresenter.kt [hunk 5]
```diff
@@ -24,47 +24,41 @@ import android.content.Context
...
 import com.saltedge.authenticator.sdk.model.response.ConfirmDenyResultData
-import com.saltedge.authenticator.sdk.model.response.isValid
 import com.saltedge.authenticator.tool.secure.fingerprint.BiometricToolsAbs
 import com.saltedge.authenticator.widget.biometric.BiometricPromptCallback
 
 abstract class BaseAuthorizationPresenter(
-    val appContext: Context,
-    val biometricTools: BiometricToolsAbs,
-    val apiManager: AuthenticatorApiManagerAbs
+    internal val appContext: Context,
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/BaseAuthorizationPresenter.kt]: We need it only in this module?

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/ViewMode.kt [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    DEFAULT,
+    CONFIRM_PROCESSING,
+    DENY_PROCESSING,
+    CONFIRM_SUCCESS,
+    DENY_SUCCESS,
+    ERROR,
+    TIME_OUT,
+    UNAVAILABLE;
+
+    fun isFinalMode(): Boolean {
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/ViewMode.kt]: mb `isValidFinalMode()`?

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/details/AuthorizationDetailsFragment.kt [hunk 7]
```diff
@@ -131,47 +124,24 @@ class AuthorizationDetailsFragment : BaseFragment(),
         headerView?.setInvisible(!show)
     }
 
-    override fun setContentVisibility(show: Boolean) {
-        contentView?.setInvisible(!show)
+    override fun setHeaderValues(logoUrl: String, title: String, startTime: DateTime?, endTime: DateTime?) {
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/details/AuthorizationDetailsFragment.kt]: Please check types of parameters

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/BaseAuthorizationPresenter.kt (hunk 5 above)
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/BaseAuthorizationPresenter.kt]: yep

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/ViewMode.kt (hunk 6 above)
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/common/ViewMode.kt]: no such concept `Valid Final Mode`

**Pull request merged**